File: JavaScript/insertion_sort.js
function insertionSort(arr) {
  for (let i = 1; i < arr.length; i++) {
    let key = arr[i];
    let j = i - 1;
    while (j >= 0 && key < arr[j]) {
      arr[j + 1] = arr[j];
      j--;
    }
    arr[j + 1] = key;
  }
}

// Exemplo de uso:
const arr = [38, 27, 43, 3, 9, 82, 10];
insertionSort(arr);
console.log(arr);

File: Swift/insertion_sort.swift
func insertionSort<T: Comparable>(_ arr: inout [T]) {
    for i in 1..<arr.count {
        let key = arr[i]
        var j = i - 1
        while j >= 0 && key < arr[j] {
            arr[j + 1] = arr[j]
            j -= 1
        }
        arr[j + 1] = key
    }
}

// Exemplo de uso:
var arr = [38, 27, 43, 3, 9, 82, 10]
insertionSort(&arr)
print(arr)
